src/test/java/duke/parser/ParserTest.java

```diff
@@ -0,0 +1,97 @@
+package duke.parser;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ParserTest {
+
+    private TaskList tasks = new TaskList();
+    private Ui ui = new Ui();
+    private Storage storage = new Storage("data/duke.txt");
+
+    @Test
+    void parse() {
+
+        // testing exception of delete
+        try {
+            Parser.parse("delete homework").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Please give me the task number in Integer.", e.getMessage());
+        }
+        try {
+            Parser.parse("delete").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Which task should I remove?", e.getMessage());
+        }
+
+        // testing exception of done
+        try {
+            Parser.parse("done homework").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Please give me the task number in Integer.", e.getMessage());
+        }
+        try {
+            Parser.parse("done").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Please give me the task number.", e.getMessage());
+        }
+
+        // testing exception of todo
+        try {
+            Parser.parse("todo").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! The description of a todo cannot be empty.", e.getMessage());
+        }
+
+        // testing exception of event
+        try {
+            Parser.parse("event").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Missing information regarding event.", e.getMessage());
+        }
+        try {
+            Parser.parse("event concert").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Missing information regarding event.", e.getMessage());
+        }
+        try {
+            Parser.parse("event /at 2020-01-27").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Missing information regarding event.", e.getMessage());
+        }
+        try {
+            Parser.parse("event concert /at 27 January 2020").execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Please give me the date in yyyy-mm-dd format!", e.getMessage());
+        }
+
+        // testing exception of deadline
```

Comment:
most of these seem like testing exceptions. what about testing the nice things, when all goes as expected?